applications/GeoMechanicsApplication/custom_processes/set_parameter_field_process.cpp

```diff
@@ -0,0 +1,155 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "set_parameter_field_process.hpp"
+
+#include <utilities/function_parser_utility.h>
+#include <utilities/mortar_utilities.h>
+
+#include "utilities/interval_utility.h"
+#include "geo_mechanics_application_variables.h"
+
+
+namespace Kratos
+{
+
+    SetParameterFieldProcess::SetParameterFieldProcess(ModelPart& rModelPart,
+                                                       const Parameters& rSettings)
+                                                            : Process(), mrModelPart(rModelPart),
+                                                            mParameters(rSettings)
+{
+    // function type: python, cpp, input
+    const Parameters default_parameters(R"(
+        {
+            "help"            : "This process applies a moving load condition belonging to a modelpart. The load moves over line elements.",
+            "model_part_name" : "please_specify_model_part_name",
+            "variable_name"   : "CUSTOM",
+            "func_type"       : "input",               
+            "function"        : "0",
+            "dataset"         : "dummy"
+        }  )"
+    );
+
+    mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+}
+
+
+void SetParameterFieldProcess::SetValueAtElement(Element& rElement, const Variable<double>& rVar, const double Value)
+{
+
+    Properties& r_prop = rElement.GetProperties();
+
+    // Copies properties
+    Properties::Pointer p_new_prop = Kratos::make_shared<Properties>(r_prop);
+
+    // Adds new properties to the element
+    p_new_prop->SetValue(rVar, Value);
+    rElement.SetProperties(p_new_prop);
+
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingInputFunction(const Variable<double>& rVar)
+{
+    auto parameter_function = BasicGenericFunctionUtility(mParameters["function"].GetString());
+    const double current_time = this->mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    for (Element& r_element : mrModelPart.Elements()) {
+
+        const auto& r_geom = r_element.GetGeometry();
+
+        // calculate parameter value at current element
+        const double val = parameter_function.CallFunction(r_geom.Center().X(), r_geom.Center().Y(), r_geom.Center().Z(), current_time, 0, 0, 0);
+
+        SetValueAtElement(r_element, rVar, val);
+
+    }
+}
+
+
+void SetParameterFieldProcess::SetParameterFieldUsingPythonFunction(const Variable<double>& rVar)
```

Comment:
To Kratos is this a string of pre-calculated values ? Thus the term PythonFunction should not be in the name, as the mParameters can be generated by any code base

Reply:
I would argue that to C++, this is SetParameterFieldUsingJsonString (So confused right now by style guide so it could be set_parameter_field_using_json_string)